image/local.go

```diff
@@ -43,15 +43,23 @@ type local struct {
 }
 
 func (f *Factory) NewLocal(repoName string, pull bool) (Image, error) {
+	var pullFailureError error
+	var inspect types.ImageInspect
+	var err error
 	if pull {
-		if err := f.pullImage(f.Out, f.Docker, repoName); err != nil {
-			return nil, fmt.Errorf("failed to pull image '%s' : %s", repoName, err)
+		err = f.pullImage(f.Out, f.Docker, repoName)
+		if err != nil {
+			pullFailureError = fmt.Errorf("failed to pull image '%s' : %s", repoName, err)
+		}
+		inspect, _ = f.inspectLocalImage(repoName)
```

Comment:
Returning an error usually implies that the other arguments have undefined contents. Check the error here, instead of checking `inspect.Config == nil`. Otherwise you miss cases where `inspect.Config != nil` and `err != nil`.